Api/Functions/SlackOutgoing.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Extensions.Logging;
+using NosAyudamos.Events;
+using Microsoft.Azure.WebJobs.Extensions.EventGrid;
+using Microsoft.Azure.EventGrid.Models;
+using System.Collections.Generic;
+using Slack.Webhooks.Elements;
+using Slack.Webhooks.Interfaces;
+using Slack.Webhooks.Blocks;
+using Slack.Webhooks;
+using Humanizer;
+using System.Text;
+using System.Net.Http;
+using System.Globalization;
+
+namespace NosAyudamos.Functions
+{
+    class SlackOutgoing : IEventHandler<UnknownMessageReceived>
+    {
+        readonly ISerializer serializer;
+        readonly IEnvironment environment;
+        readonly IPersonRepository repository;
+        readonly ILanguageUnderstanding language;
+        readonly HttpClient http;
+        readonly ILogger<SlackOutgoing> logger;
+
+        public SlackOutgoing(ISerializer serializer, IEnvironment environment, IPersonRepository repository, ILanguageUnderstanding language, HttpClient http, ILogger<SlackOutgoing> logger) =>
+            (this.serializer, this.environment, this.repository, this.language, this.http, this.logger) = (serializer, environment, repository, language, http, logger);
+
+        [FunctionName("slack_outgoing")]
+        public Task SendAsync([EventGridTrigger] EventGridEvent e) => HandleAsync(e.GetData<UnknownMessageReceived>(serializer));
+
+        public async Task HandleAsync(UnknownMessageReceived e)
```

Comment:
maybe HandleUnknownIntentAsync?

Reply:
The method name is part of `IEventHandler<T>`, can't change it ;)

Reply:
I mean SendAsync -> HandleUnknownIntentAsync